diff --git a/ports/mpfr/portfile.cmake b/ports/mpfr/portfile.cmake
index 219b92d13..864e78ebd 100644
--- a/ports/mpfr/portfile.cmake
+++ b/ports/mpfr/portfile.cmake
@@ -14,9 +14,17 @@ vcpkg_extract_source_archive(
         src-only.patch
 )
 
+vcpkg_list(SET OPTIONS)
+
+if(VCPKG_TARGET_IS_MINGW AND VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic")
+    vcpkg_list(APPEND OPTIONS "--disable-static --enable-shared")
+endif()
+
 vcpkg_configure_make(
     SOURCE_PATH "${SOURCE_PATH}"
     AUTOCONFIG
+    OPTIONS
+        ${OPTIONS}
 )
 
 vcpkg_install_make()
